packages/core/src/common/messaging/web-socket-channel.ts

```diff
@@ -93,6 +93,20 @@ export class WebSocketChannel implements IWebSocket {
         this.fireClose(code, reason);
     }
 
+    tryClose(code: number = 1000, reason: string = ''): void {
```

Comment:
The only difference with `close` is that this one doesn't throw. Is it really required?

We could make `close` not throw anymore? No internal client is actually wrapping it in a throw/catch clause either...

I am fine with the current change, but I am curious.

Reply:
If I use `close` instead of `tryClose` in BaseLanguageServerContribution, an "already disposed" error is thrown occasionally on page reload.

I think it's fine for `close` to throw on disposed because it's expected to send a message to the other endpoint. We could also add another optional parameter instead of a separate method, though.